pkgs/tools/typesetting/lowdown/default.nix

```diff
@@ -14,6 +14,8 @@ stdenv.mkDerivation rec {
   nativeBuildInputs = [ which ]
     ++ lib.optionals stdenv.isDarwin [ fixDarwinDylibNames ];
 
+  NIX_CFLAGS_COMPILE = lib.optional stdenv.hostPlatform.isMusl "-fPIC";
```

Comment:
This is not precise enough `stdenv.hostPlatform.isMusl` is `true` for `pkgsStatic` also which is working fine. I hope we can fix the issue via the configure script somehow?!

Reply:
Hmm, good point. Perhaps a more correct approach is to compile with -fPIC whenever a .so will be built, i.e. when !stdenv.hostPlatform.isStatic.

I don't think this would enable PIC anywhere it wasn't before because:
* the `pic` hardening flag has been implicitly adding -fPIC for all currently gnu64 builds
* afaik PIC is always needed to build a shared library

Does that sound correct?

Reply:
Right, I remember now that we've hacked in building a dynamic library into lowdown some time ago (in 0ceadf99f221cf461efa9edb34424ff6ad375119, this was requested by nix upstream), thus it is not surprising that lowdown's configure system doesn't handle this case  correctly.

The best solution, I think, would be to amend `shared.patch` so that it adds `-fPIC` to `CFLAGS` in the `Makefile`, since that patch is already included conditionally at the right time.

Reply:
Ok yeah that seems cleanest, pushed